tracer/test/Datadog.Trace.TestHelpers/ExitCodeException.cs

```diff
@@ -0,0 +1,23 @@
+// <copyright file="ExitCodeException.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+
+namespace Datadog.Trace.TestHelpers;
+
+public class ExitCodeException : Exception
+{
+    public ExitCodeException(int actualExitCode, int expectedExitCode, string message)
+        : base(string.IsNullOrWhiteSpace(message) ?
+                   $"Expected exit code: {expectedExitCode}, actual exit code: {actualExitCode}." :
+                   $"Expected exit code: {expectedExitCode}, actual exit code: {actualExitCode}. Message: {message}")
+    {
+    }
+
+    public static void Throw(int actualExitCode, int expectedExitCode, string message = null)
+    {
+        throw new ExitCodeException(actualExitCode, expectedExitCode, message);
+    }
```

Comment:
Nice, though I would probably create a helper to remove all those `if`s
```suggestion
    public static void ThrowIfNonZero(int actualExitCode, string message = null)
        => Throw(actualExitCode, expectedExitCode: 0, message);


    public static void Throw(int actualExitCode, int expectedExitCode, string message = null)
        =>  throw new ExitCodeException(actualExitCode, expectedExitCode, message);
```

Reply:
done in [73d791c](https://github.com/DataDog/dd-trace-dotnet/pull/3425/commits/73d791cac6d07899dc549801ab50b3276bb6680c)